lib/immediate-juttle-job.js

```diff
@@ -0,0 +1,96 @@
+var _ = require('underscore');
+var logger = require('log4js').getLogger('juttle-job');
+var JuttleJob = require('./juttle-job');
+
+var ImmediateJuttleJob = JuttleJob.extend({
+    initialize: function(options) {
+        var self = this;
+
+        self._timeout = options.timeout || 60000;
+
+        // The output of this program, as a hash sink_id -> sink
+        // description + array of points.
+        self._program_output = {};
+
+        // Any errors for this program.
+        self._errors = [];
+
+        // Any warnings for this program.
+        self._warnings = [];
+    },
+
+    send_to_clients: function(msg) {
```

Comment:
Something about having subclasses override `send_to_clients` feels odd to me. Two concrete reasons I can come up with is that in this specific `send_to_clients` implementation there are actually no clients, and generalizing that a bit more, it seems like whoever is running the `JuttleJob` should be responsible for handling the clients, not the job itself.

Perhaps it would make more sense for `ImmediateJuttleJob` and `WebsocketJuttleJob` to _use_ a `JuttleJob` instead of extending it. That way they could listen to the messages and do w/e they need to (either store in an array in the case of `ImmediateJuttleJob` or do endpoint-y stuff in the case of `WebsocketJuttleJob`).

Thoughts?